libs/template/renderer.go

```diff
@@ -112,6 +119,16 @@ func (r *renderer) executeTemplate(templateDefinition string) (string, error) {
 	result := strings.Builder{}
 	err = tmpl.Execute(&result, r.config)
 	if err != nil {
+		// Parse and return a more readable error for missing values that are used
+		// by the template definition but are not provided in the passed config.
+		if strings.Contains(err.Error(), "map has no entry for key") {
+			captureRegex := regexp.MustCompile(`map has no entry for key "(.*)"`)
+			matches := captureRegex.FindStringSubmatch(err.Error())
+			if len(matches) != 2 {
+				return "", err
+			}
+			return "", fmt.Errorf("variable %q not defined", matches[1])
+		}
```

Comment:
Please investigate unwrapping `template.ExecError` instead of manipulating its string version.

This should only happen on `errors.As` one of those and then you can manipulate the message field directly.

See https://pkg.go.dev/text/template#ExecError

Reply:
Agree with Pieter here, just for reference in case it's useful
https://cs.opensource.google/go/go/+/master:src/text/template/exec.go;l=688?q=%22map%20has%20no%20entry%20for%20key%22&ss=go%2Fgo